### DISCUSSION TITLE
Any recommendation to fix no method matching pdf?

### DISCUSSION THREAD

**Discussion by Etwari3** [JuliaPOMDP/POMDPs.jl#389] [Debugging Help]
MethodError: no method matching pdf(::Array{Float64, 3}, ::String)

**Comment by zsunberg** [comment]
This means that the solver is trying to evaluate the probability of a particular string in an array of floats.

This error is likely caused by returning an array of Floats instead of a [Distribution object](http://juliapomdp.github.io/POMDPs.jl/stable/interfaces/) in `transition`, `observation` or `initialstate`. Judging by the error, it seems like your states are `String`s. So instead of something like
```julia
return [0.1, 0.2, 0.7]
```
you should use a distribution object such as a ['SparseCat'](https://juliapomdp.github.io/POMDPModelTools.jl/stable/distributions/#Sparse-Categorical-(SparseCat))
```julia
return SparseCat(["a", "b", "c"], [0.1, 0.2, 0.7])
```
in your `transition`, `observation`, and `initialstate` functions.

Side note: for future questions, it may be more helpful to post in the discussions section of POMDPs.jl https://github.com/JuliaPOMDP/POMDPs.jl/discussions rather than as an issue in this particular solver.

**Reply by zsunberg** [reply]
(I migrated this to a POMDPs.jl issue to concentrate as much knowledge as possible)

**Reply by Etwari3** [reply]
Thank you. 

My apologies for the misplacement of the question.

**Reply by Etwari3** [reply]
Converted all (transition, observation, and initialstates function) to SparseCat and still did not work.

What does POMDPs.pdf function is used for? I noticed it in both 'Simple Grid World' model and the fixhorizon.jl. I could not find it in the documentation.

Thank you.

**Reply by zsunberg** [reply]
It is the same as `Distributions.pdf`: https://juliapomdp.github.io/POMDPs.jl/stable/api/#Distributions.pdf

It is used by solvers to evaluate the probability of a transition or observation outcome.

e.g.
```julia
distribution = SparseCat(["a", "b"], [0.1, 0.9])
pdf(distribution, "a") # should return 0.1
```

**Comment by Etwari3** [comment]
Thank you.
